### PR TITLE
tests: failure of prereqs on content interface doesn't prevent install

### PR DESCRIPTION
**Pull request by stolowski**
Related to https://forum.snapcraft.io/t/snap-left-in-unusable-state-after-failure-to-refresh/24253, this PR reproduces this.

~~It seems that due to usage of lanes to install prerequisites, we install the snap even if its prereqs fail to download (at least for content interface prereqs, I'm not sure about bases yet).~~

As discussed this works as designed, no changes planned wrt content-interface dependencies. I think this PR is ok to land for extra unit tests.

### CODE REVIEW

**Comment by stolowski** [comment]
Per discussion on IRC this is expected behavior as content interface defines "soft" dependencies. I think this PR should be ok to land to have extra tests, and they can be useful in case we want to work on a solution for the issue.

File: overlord/snapstate/snapstate_test.go [hunk 1]
```diff
@@ -5300,93 +5301,94 @@ func (s contentStore) SnapAction(ctx context.Context, currentSnaps []*store.Curr
 	if err != nil {
 		return nil, nil, err
 	}
-	if len(sars) != 1 {
-		panic("expected to be queried for install of only one snap at a time")
-	}
```
Comment (bboozzoo) [inline overlord/snapstate/snapstate_test.go:5305]: should this check be restored?

**Review by bboozzoo — APPROVED** [review]
LGTM

File: overlord/snapstate/snapstate_test.go (hunk 1 above)
Comment (stolowski) [reply overlord/snapstate/snapstate_test.go]: Yeah that's fine, although it needs to be relaxed now and allow 1 or 2 actions. Updated, thanks.

**Comment by stolowski** [comment]
I discussed this with @pedronis and it's actually "by design" right now. We don't have a concept of required prerequisites, so everything is optional. I think it still makes sense to have an explicit test (as a starting point for potential improvements in the future).

**Review by MiguelPires — COMMENTED** [review]
LGTM. Just a small comment regarding the slice sort

**Pull request merged**